ashpd-backend-demo/src/main.rs

```diff
@@ -0,0 +1,124 @@
+use std::{cell::RefCell, rc::Rc, sync::Arc};
+
+use ashpd::backend::Backend;
+use gettextrs::LocaleCategory;
+use gtk::glib;
+//#[allow(deprecated)] // We use the deprecated GTK APIs
+// mod file_chooser;
+mod account;
+mod account_preview;
+mod screenshot;
+mod settings;
+mod wallpaper;
+mod wallpaper_preview;
+
+// use file_chooser::FileChooser;
+use account::Account;
+use screenshot::Screenshot;
+use settings::Settings;
+use wallpaper::Wallpaper;
+
+// NOTE Uncomment if you have ashpd-backend-demo.portal installed.
+const NAME: &str = "org.freedesktop.impl.portal.desktop.ashpd-backend-demo";
+// const NAME: &str = "org.freedesktop.impl.portal.desktop.gnome";
+
+fn main() {
+    // Enable debug with `RUST_LOG=xdp_ashpd_gnome=debug COMMAND`.
+    tracing_subscriber::fmt::init();
+
+    // FIXME Use meson here
+    gettextrs::setlocale(LocaleCategory::LcAll, "");
+    gettextrs::bindtextdomain("ashpd-backend-demo", "/usr/share/locale")
+        .expect("Unable to bind the text domain");
+    gettextrs::textdomain("ashpd-backend-demo").expect("Unable to switch to the text domain");
+
+    glib::set_prgname(Some("ashpd-backend-demo"));
+
+    // Avoid pointless and confusing recursion
+    glib::unsetenv("GTK_USE_PORTAL");
+    glib::setenv("ADW_DISABLE_PORTAL", "1", true).unwrap();
+    glib::setenv("GSK_RENDERER", "cairo", true).unwrap();
```

Comment:
This has really really bad performance on gtk4 in GNOME 44 so should be avoided. With that said, it's not possible to have GdkDisplays with different backends while still using hardware acceleration. See https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/68.

Reply:
I am confused by this, we need x11 displays to have modal windows on windows using xwalyand. Does this means that xdp-gnome does not do modals on them anymore? By the way note that initializing gtk with the GL backaend takes extremely long (~0.5s-1s on my machines), not a big problem if it is only done once but still.

Reply:
xdp-gnome 44 uses Wayland for modals on top of both Wayland and X11 windows if it's a Wayland session, and X11 otherwise. It uses a [custom protocol](https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/blob/main/data/mutter-x11-interop.xml) just for that.